email/parser.go

```diff
@@ -30,16 +30,21 @@ const (
 )
 
 var (
-	// extractSkylink64RE is a regex that is capable of extracting base-64
-	// encoded skylinks from text
-	extractSkylink64RE = regexp.MustCompile(".+?://.+?\\..+?/([a-zA-Z0-9-_]{46})")
+	// extractSkylink64RE and extractSkylink64RE_2 are regexes capable of
+	// extracting base-64 encoded skylinks from text
+	extractSkylink64RE   = regexp.MustCompile(`.+?://.+?\..+?/([a-zA-Z0-9-_]{46})`)
+	extractSkylink64RE_2 = regexp.MustCompile(`(http.+|hxxp.+|\..+|://.+|^)([a-zA-Z0-9-_]{46})(\?.*)?$`)
 
-	// extractSkylink32RE is a regex that is capable of extracting base-32
-	// encoded skylinks from text
-	extractSkylink32RE = regexp.MustCompile(".+?://.*?([a-zA-Z0-9-_]{55})")
+	// extractSkylink32RE and extractSkylink32RE_2 are regexes capable of
+	// extracting base-32 encoded skylinks from text
+	extractSkylink32RE   = regexp.MustCompile(`.+?://.*?([a-zA-Z0-9-_]{55})`)
+	extractSkylink32RE_2 = regexp.MustCompile(`(http.+|hxxp.+|\..+|://.+|^)([a-zA-Z0-9-_]{55})(\?.*)?$`)
 
-	validateSkylink64RE = regexp.MustCompile("^([a-zA-Z0-9-_]{46})$")
-	validateSkylink32RE = regexp.MustCompile("^([a-zA-Z0-9-_]{55})$")
+	// space matches all whitespace
+	space = regexp.MustCompile(`\s+`)
+
+	validateSkylink64RE = regexp.MustCompile(`^([a-zA-Z0-9-_]{46})$`)
+	validateSkylink32RE = regexp.MustCompile(`^([a-zA-Z0-9-_]{55})$`)
```

Comment:
base32 is just letters and numbers (adding case insensitive flag because some emails may have urls in all caps)

```suggestion
	validateSkylink32RE = regexp.MustCompile(`(?i)^([a-z0-9]{55})$`)
```

Reply:
updated